monitor/api/alerts.py

```diff
@@ -40,21 +42,50 @@ def _query_alert_type(self, run_operation_args: dict, alert_factory_func: Callab
         if raw_alerts:
             alert_dicts = json.loads(raw_alerts[0])
             for alert_dict in alert_dicts:
+                notmalaized_alert = self._normalize_alert(alert=alert_dict)
                 try:
                     alerts.append(alert_factory_func(
                         elementary_database_and_schema=self.elementary_database_and_schema,
-                        **alert_dict
+                        **notmalaized_alert
                     ))
                 except Exception:
                     malformed_alerts.append(MalformedAlert(
-                        alert_dict['id'],
-                        self.elementary_database_and_schema,
-                        alert_dict,
+                        id=notmalaized_alert['id'],
+                        elementary_database_and_schema=self.elementary_database_and_schema,
+                        data=notmalaized_alert,
+                        slack_channel=notmalaized_alert.get('slack_channel')
                     ))
         if malformed_alerts:
             logger.error('Failed to parse some alerts.')
             self.success = False
         return AlertsQueryResult(alerts, malformed_alerts)
+    
+    @classmethod
+    def _normalize_alert(cls, alert: dict) -> dict:
+        normalized_alert = copy.deepcopy(alert)
```

Comment:
why not doing it in place? deepcopy is slow

Reply:
I hate mutables objects.
There are more chances for bugs and its just not worth it IMO.
I can change it if you want.
Deepcopy is slow, but we have small objects so I don't mind using it 🤷🏻‍♂️